docs/design/chart_format.md

```diff
@@ -0,0 +1,664 @@
+
+
+The combination of this gives users the following assurances:
+
+
+* The images this chart references at build time are still the same exact version when installed (checksum images).
+   * This is distinct from asserting that the image Kubernetes is running is exactly the same version that a chart references. Kubernetes does not currently give us a way of verifying this.
+* The package itself has not been tampered with (checksum package tgz).
+* The entity who released this package is known (via the GPG/PGP signature).
+
+
+The format of the file is as follows:
+
+```
+-----BEGIN PGP SIGNED MESSAGE-----
+name: nginx
+description: The nginx web server as a replication controller and service pair.
+version: 0.5.1
+keywords:
+  - https
+  - http
+  - web server
+  - proxy
+source: https://github.com/foo/bar
+home: http://nginx.com
+depends:
+        kubernetes:
+                version: >= 1.0.0
+---
+files:
+        nginx-0.5.1.tgz: “sha256:9f5270f50fc842cfcb717f817e95178f”
+images:
+        “hub.docker.com/_/nginx:5.6.0”: “sha256:f732c04f585170ed3bc99”
+-----BEGIN PGP SIGNATURE-----
+Version: GnuPG v1.4.9 (GNU/Linux)
+
+
+iEYEARECAAYFAkjilUEACgQkB01zfu119ZnHuQCdGCcg2YxF3XFscJLS4lzHlvte
+WkQAmQGHuuoLEJuKhRNo+Wy7mhE7u1YG
+=eifq
+-----END PGP SIGNATURE-----
+```
+
+Note that the YAML section contains two documents (separated by ---\n). The first is the Chart.yaml. The second is the checksums, defined as follows.
+
+
+* Files: A map of filenames to SHA-256 checksums (value shown is fake/truncated)
+* Images: A map of image URLs to checksums (value shown is fake/truncated)
+
+
+The signature block is a standard PGP signature, which provides [tamper resistance](http://www.rossde.com/PGP/pgp_signatures.html).
+
+### Repository Protocol
+A repository must implement HTTP(S) GET operations on both chart names (nginx-1.2.3) and chart signatures (nginx-1.2.3.sig). HTTP GET is a rational base level of functionality because it is well understood, nearly ubiquitous, and simple.
+
+
+A repository may implement the full Object Storage APIs for compatibility with S3 and Google Cloud Storage. In this case, a client may detect this and use those APIs to perform create, read, update, delete, and list operations (as permissions allow) on the object storage system.
+
+
+Object storage is a rational choice for this model because it is [optimized for highly resilient, distributed, highly available read-heavy traffic](https://en.wikipedia.org/wiki/Object_storage). The S3-like API has become a de facto standard, with myriad client libraries and tools, and major implementations both as services (S3, GCS), platforms (OpenStack Swift), and individual stand-alone servers (RiakCS, Minio, Ceph).
+
+
+This proposal does not mandate a particular authentication mechanism, but implementors may implement the authentication token mechanisms on GCS and S3 compatible object storage systems.
```

Comment:
We could say that we allow no authentication or token based authentication. Would that be okay?